catroid/src/main/java/org/catrobat/catroid/io/DeviceVariableManager.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.io;
+
+import android.util.Log;
+
+import com.parrot.freeflight.utils.FileUtils;
+
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Scene;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.UUID;
+
+import static org.catrobat.catroid.common.Constants.DEVICE_VARIABLE_DIR;
+
+public final class DeviceVariableManager {
```

Comment:
About this - it's yet another manager class and yet another singleton.
`Manager` most of the time means that it does way more than it should.

From what I'm seeing here this actually just manages a global lock via the `synchronized` keyword.
Other than that it just hides the fact that everything in here are public static helper methods.

There should be a more object oriented way to solve this.